code/modules/projectiles/updated_projectiles/ammo_datums.dm

```diff
@@ -143,21 +145,26 @@
 				for(var/i=0, i<knockback, i++)
 					step_away(M,P)
 
-		//Check for and apply soft CC; Xeno only at this time
-		if(isXeno(M) && (M.mob_size == MOB_SIZE_BIG && soft_size_threshold > 2) || (M.mob_size == MOB_SIZE_XENO && soft_size_threshold > 1))
-			var/mob/living/carbon/Xenomorph/X = M
+		//Check for and apply soft CC
+		if(iscarbon(M))
+			var/mob/living/carbon/C = M
+			var/stagger_immune = FALSE
+			if(isXeno(C))
+				var/mob/living/carbon/Xenomorph/X = M
+				if(isXenoQueen(X)) //Stagger too powerful vs the Queen, so she's immune.
```

Comment:
Give them a stagger immunity flag instead of this.

Reply:
This is as it was before essentially.